api/src/main/java/me/devnatan/inventoryframework/ViewConfigBuilder.java

```diff
@@ -6,149 +6,162 @@
-     */
-    public ViewConfigBuilder cancelOnPickup() {
-        return addOption(ViewConfig.CANCEL_ON_PICKUP);
-    }
-
-    /**
-     * Cancels any item drops by the player while the view is open.
-     *
-     * @return This configuration builder.
-     */
-    public ViewConfigBuilder cancelOnDrop() {
-        return addOption(ViewConfig.CANCEL_ON_DROP);
-    }
-
-    public ViewConfigBuilder cancelOnDrag() {
-        throw new UnsupportedOperationException();
-    }
-
-    public ViewConfig build() {
-        final Map<ViewConfig.Option<?>, Object> optionsMap = options.stream()
-                .map(option -> new AbstractMap.SimpleImmutableEntry<ViewConfig.Option<?>, Object>(
-                        option, option.defaultValue()))
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-
-        return new ViewConfig(title, size, type, optionsMap, layout, modifiers);
-    }
+	private String title = "";
+	private int size = 0;
+	private ViewType type;
+	private final Set<ViewConfig.Option<?>> options = new HashSet<>();
+	private String[] layout = null;
+	private final Set<LayoutSlot> patterns = new HashSet<>();
+	private final Set<ViewConfig.Modifier> modifiers = new HashSet<>();
+
+	/**
+	 * Inherits all configuration from another {@link ViewConfigBuilder} value.
+	 * <p>
+	 * Note that the values will be merged and not replaced, however, the values of the setting to
+	 * be inherited take precedence over those of that setting.
+	 *
+	 * @param other The configuration that will be inherited.
+	 * @return This config.
+	 */
+	public ViewConfigBuilder inheritFrom(@NotNull ViewConfigBuilder other) {
+		throw new UnsupportedOperationException("Inheritance is not yet supported");
+	}
+
+	/**
+	 * Defines the type of the container.
+	 * <p>
+	 * If applied in view scope, it will be the default value for all contexts originated from it.
+	 *
+	 * @param type The container type.
+	 * @return This config.
+	 */
+	public ViewConfigBuilder type(ViewType type) {
+		this.type = type;
+		return this;
+	}
+
+	/**
+	 * Defines the title of the container.
+	 * <p>
+	 * If applied in view scope, it will be the default value for all contexts originated from it.
+	 *
+	 * @param title The container title.
+	 * @return This configuration builder.
+	 */
+	public ViewConfigBuilder title(String title) {
+		this.title = title;
+		return this;
+	}
+
+	/**
+	 * Defines the size of the container.
+	 * <p>
+	 * If applied in view scope, it will be the default value for all contexts originated from it.
+	 *
+	 * @param size The container size.
+	 * @return This configuration builder.
+	 * @throws InvalidSizeException If used after setting a layout and size differs from the layout
+	 *                              length.
+	 */
+	public ViewConfigBuilder size(int size) {
+		if (layout != null && size != layout.length) {
+			throw new InvalidSizeException("The size differs from the set layout length.");
+		}
```

Comment:
this check must be in the OpenInterceptor [on container creation](https://github.com/DevNatan/inventory-framework/blob/2ae01ae6df7e42729cbe136a5a7949859ea3bae5/platform/src/main/java/me/devnatan/inventoryframework/pipeline/OpenInterceptor.java#L73) since both layout and container size can be dynamic, and, not necessarily, the layout of this ViewConfigBuilder is the layout that will be effectively used